timestampvm/vm.go

```diff
@@ -210,17 +275,16 @@ func (vm *VM) proposeBlock(data [dataLen]byte) {
 // from another node
 func (vm *VM) ParseBlock(bytes []byte) (snowman.Block, error) {
 	// A new empty block
-	block := &Block{}
+	block := &TimeBlock{}
 
 	// Unmarshal the byte repr. of the block into our empty block
-	_, err := vm.codec.Unmarshal(bytes, block)
+	_, err := Codec.Unmarshal(bytes, block)
 	if err != nil {
 		return nil, err
 	}
 
 	// Initialize the block
-	// (Block inherits Initialize from its embedded *core.Block)
-	block.Initialize(bytes, &vm.SnowmanVM)
+	block.Initialize(bytes, choices.Processing, vm)
```

Comment:
I feel like there could be a bug here - relating to if a block that was previously accepted is parsed, then it's status will be `Processing`. We should probably try to simplify the invariants of statuses.

Reply:
not sure how we can simplify it. maybe instead of holding the status in block we can depend on state to report it? i.e 
```
func (b *TimeBlock) Status() choices.Status { 
    return b.vm.state.GetStatus(b.ID()) 
}
```

currently I fetched it from state as we do it other VMs.